### PR TITLE
Added support for summons tag "max"

### PR DESCRIPTION
**Pull request by raymondtfr**
This was discussed in #1905.

This code was originally from a different distribution of TFS.

I just updated it use a range-based for loop and also renamed a variable to be more clear.

I have tested it and it works fine.

### CODE REVIEW

**Comment by ranisalt** [comment]
I wouldn't use "amount" as it refers to a current value rather than a maximum value. It's even possible to use "maximum amount". "max" is fine.

**Comment by marksamman** [comment]
It should also be uint32_t when parsing it, and default to the max value (or 0 which would mean no limit) so that the current behavior of not specifying max is intact.

**Comment by raymondtfr** [comment]
You mean renaming both the summonBlock_t data member and the other "amount" to "max"?

**Comment by ranisalt** [comment]
> You mean renaming both the summonBlock_t data member and the other "amount" to "max"?

Yep, "amount" does not have the meaning you want. "max" is the correct word here.

File: src/monster.cpp [hunk 1]
```diff
@@ -942,6 +942,17 @@ void Monster::onThinkDefense(uint32_t interval)
 				continue;
 			}
 
+			uint32_t summonCount = 0;
+			for (Creature* summon : summons) {
+				if (!summon->isRemoved() && summon->getName() == summonBlock.name) {
```
Comment (marksamman) [inline src/monster.cpp]: You can remove the isRemoved() check since we use summons.size() to check for maxSummons above.

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**